### PR TITLE
[chrome] Add DevTools Recorder

### PR DESCRIPTION
**Pull request by nickserv**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.chrome.com/docs/extensions/reference/devtools_recorder/

### CODE REVIEW

**Comment by nickmccurdy** [comment]
Depends on https://github.com/microsoft/DefinitelyTyped-tools/pull/525

**Comment by nickmccurdy** [comment]
`"target": "es6"` doesn't seem to be working but I'm not sure why

**Comment by nickmccurdy** [comment]
I'd appreciate help with the build errors.

**Comment by JasonXian** [comment]
Hey @nickmccurdy, I played around with different tsconfig options but couldn't seem to solve the issue either. I agree that updating the target to es6 should fix the puppeteer errors but it seems to just generates other errors as well.

For now, instead of importing `Step` and `UserFlow` for the argument types, it might be simpler to use `any` for now. I'd assume any downstream use cases would be passing in a properly typed argument from the types in the original library anyways, meaning there would still be some safety.

This isn't an ideal solution, but I don't think this build system issue is worth being blocked on in the meanwhile.

File: types/chrome/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "private": true,
+  "dependencies": {
+    "@puppeteer/replay": "^1.0.0"
```
Comment (fregante) [inline types/chrome/package.json:4]: I'm not a fan of adding [1MB-plus-dependency](https://packagephobia.com/result?p=%40puppeteer%2Freplay) to a type package. I think this should be dropped. I think that if someone needs those types they can install it independently, even if this d.ts explicitly mentions `import type {Step, UserFlow} from '@puppeteer/replay'`

If that's not acceptable, how about venturing replay’s d.ts instead? https://unpkg.com/browse/@puppeteer/replay@2.9.0/lib/main.d.ts

**Review by fregante — CHANGES_REQUESTED** [review]

**Pull request closed without merging**